streams/src/test/java/org/apache/kafka/streams/tools/StreamsResetterTest.java

```diff
@@ -237,12 +235,10 @@ public void shouldSeekToEndOffset() {
     @Test
     public void shouldDeleteTopic() {
         Cluster cluster = createCluster(1);
-        try (MockKafkaAdminClientEnv env = new MockKafkaAdminClientEnv(cluster)) {
-            env.kafkaClient().setNode(cluster.controller());
-            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
-            env.kafkaClient().prepareMetadataUpdate(env.cluster(), Collections.<String>emptySet());
-            env.kafkaClient().prepareResponse(new DeleteTopicsResponse(Collections.singletonMap(TOPIC, Errors.NONE)));
-            streamsResetter.doDelete(Collections.singletonList(topicPartition.topic()), env.adminClient());
+        try (MockAdminClient adminClient = new MockAdminClient(cluster.nodes())) {
+            TopicPartitionInfo topicPartitionInfo = new TopicPartitionInfo(0, cluster.nodeById(0), cluster.nodes(), Collections.<Node>emptyList());
+            adminClient.addTopic(false, TOPIC, Collections.singletonList(topicPartitionInfo), null);
+            streamsResetter.doDelete(Collections.singletonList(TOPIC), adminClient);
         }
```

Comment:
I am wondering why this test does not verify that the topic was deleted... I guess we should add this.

Reply:
Yes, adding it.